proposals/Fleet-Windows-OS-vulns.md

```diff
@@ -0,0 +1,139 @@
+# Detecting Windows OS vulnerabilities
+
+The first step in detecting Windows vulnerabilities is understanding how they are remediated. Using
+the CISA list of know vulnerabilities as a sample, it was determined that all vulnerability
+remediations follow the form of *applying some software patch* and then *following some
+steps (if any)* so it follows that to determine if a system is susceptible to a vulnerability we will need to
+check whether a specific software patch was applied and also if the provided steps were followed. So
+basically we have three problems we need to solve: 
+1. We need a list of vulnerabilities + remediations. Each remediation is composed of a patch + some steps.
+2. For a given vulnerability we need to determine whether the proper patch was applied.
+3. For a given vulnerability we need to determine whether the steps (if any) were followed.
+
+## List of vulnerabilities
+
+To get the list of vulnerabilities we can use the [Microsoft Security Updates API](https://api.msrc.microsoft.com/cvrf/v2.0/swagger/index). This [endpoint](api.msrc.microsoft.com/cvrf/v2.0/document/yyyy-mm) in particular exposes all the security updates released for yyyy-mmm formatted according to the [Common Vulnerability Reporting Framework](http://docs.oasis-open.org/csaf/csaf-cvrf/v1.2/csaf-cvrf-v1.2.html#:~:text=The%20CSAF%20Common%20Vulnerability%20Reporting,impact%20and%20remediation%20among%20interested) (cvrf) format, this cvrf document will include entries for vulnerabilities, how to patch them and what products are affected. 
```

Comment:
TIL about this, thank you! Seems like the source code is open source, and you can even use it directly from PowerShell! https://github.com/microsoft/MSRC-Microsoft-Security-Updates-API/blob/d0652effcddec621322a673382a5786f3ce2bd8b/src/README.md